### PR TITLE
agent: Fix execute_hook() args error

### PR DESCRIPTION
**Pull request by Bevisy**
The hook.args[0] is the binary name which shouldn't be included
in the Command.args.

refer [posix-platform-hooks](https://github.com/opencontainers/runtime-spec/blob/main/config.md#posix-platform-hooks)

Fixes: #2610

Signed-off-by: Binbin Zhang <binbin36520@gmail.com>

### CODE REVIEW

**Comment by Bevisy** [comment]
find_hooks() https://github.com/kata-containers/kata-containers/blob/0fe20854e7e34199a6acec68f24e2ea91597eff5/src/agent/src/sandbox.rs#L333

File: src/agent/rustjail/src/container.rs
Comment (jodh-intel) [inline src/agent/rustjail/src/container.rs:1663]: Looking at the existing unit tests, I don't see one that correctly runs a hook and checks the hook output (the `xargs` test doesn't really count imho). That probably explains why this feature has been broken too! ;)

Therefore, to prove the fix on this PR works please can you add a new test that does the equivalent of something like this:

```bash
tmpdir=$(mktemp -d)
touch ${tmpdir}/a ${tmpdir}/b ${tmpdir}/c
```

The test should then check to ensure all three files are created.

**Review by jodh-intel — COMMENTED** [review]
Thanks @Bevisy! I've made a suggestion for a new test that would help here I think.

File: src/agent/rustjail/src/container.rs
Comment (Bevisy) [reply src/agent/rustjail/src/container.rs:1663]: Thanks for your advice, I will do some fixes about tests when I figured out other problems I found related to using hook in guest vm.

**Comment by liubin** [comment]
@Bevisy any update?

**Comment by Bevisy** [comment]
PTAL @jodh-intel @liubin 
The failure of ci-s390x does not seem to be related to this modification.

File: src/agent/rustjail/src/container.rs
Comment (liubin) [inline src/agent/rustjail/src/container.rs:1548]: @Bevisy you added reading stderr code here, so can you also add a test case that runs invalid hooks and generate stderr?

Comment (Bevisy) [reply src/agent/rustjail/src/container.rs:1548]: Thank you for your suggestion, I will deal with it later

Comment (Bevisy) [reply src/agent/rustjail/src/container.rs:1548]: I have add the test which can redirect logs to stderr, but this test looks awkward. Because it looks like doing nothing. Can you give any advice?

**Comment by Bevisy** [comment]
PTAL @liubin @jodh-intel

File: src/agent/rustjail/src/container.rs
Comment (liubin) [inline src/agent/rustjail/src/container.rs:1667]: It's better and safe to move this `defer` before `execute_hook`.

File: src/agent/rustjail/src/container.rs
Comment (liubin) [inline src/agent/rustjail/src/container.rs:1724]: `execute_hook` is calld but result is not checked, `Ok` or `Err`.

**Comment by Bevisy** [comment]
Hold on until the issue https://github.com/kata-containers/tests/issues/4388 fixed, or ci can't pass.

**Comment by Bevisy** [comment]
@Tim-Zhang Do you have any idea about the ci failure?
```sh
thread 'container::tests::test_execute_hook' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }', rustjail/src/container.rs:1532:14
```

File: src/agent/rustjail/src/container.rs
Comment (liubin) [inline src/agent/rustjail/src/container.rs:1556]: Since below will also log an error message, here `error!` can be replaced with `info!`.

**Review by liubin — DISMISSED** [review]

File: src/agent/rustjail/src/container.rs
Comment (Bevisy) [reply src/agent/rustjail/src/container.rs:1556]: Ok

File: src/agent/rustjail/src/container.rs
Comment (Jakob-Naucke) [inline src/agent/rustjail/src/container.rs:1647]: Can you `let` the command once? (Also for the ones below.) LGTM otherwise.

Comment (Bevisy) [reply src/agent/rustjail/src/container.rs:1647]: Sorry, I don't understand what once means here, can you say it more clearly or give an example?

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

File: src/agent/rustjail/src/container.rs
Comment (Jakob-Naucke) [reply src/agent/rustjail/src/container.rs:1647]: Sorry, I was very brief. I meant something like
```rust
let command = "touch";
let touch = which(command).await;
…
    args: vec![command.to_string(), …
```

etc.

**Review by bergwolf — APPROVED** [review]
lgtm!

**Review by fidencio — APPROVED** [review]
lgtm, thanks @Bevisy!

**Pull request merged**
